File: server/model/notification.go
package model

import "time"

const (
	NotificationDefaultIcon  = "PhBellRinging"
	NotificationDefaultColor = "#64748B"
)

type Notification struct {
	ID                 int64      `json:"id" gorm:"type:bigint;primaryKey;autoIncrement"`
	Type               string     `json:"type" gorm:"size:32;not null"`
	AudienceType       string     `json:"audience_type" gorm:"size:20;not null;default:user;index:idx_audience_status_publish,priority:1"`
	ReceiverID         *int       `json:"receiver_id" gorm:"type:int;index:idx_receiver_status_publish,priority:1"`
	SenderID           *int       `json:"sender_id" gorm:"type:int"`
	Title              string     `json:"title" gorm:"size:255;not null"`
	Content            string     `json:"content" gorm:"type:text;not null"`
	IconName           string     `json:"icon_name" gorm:"size:64;not null;default:PhBellRinging"`
	IconColor          string     `json:"icon_color" gorm:"size:16;not null;default:#64748B"`
	SourceType         string     `json:"source_type" gorm:"size:32;index:idx_source,priority:1"`
	SourceID           *int64     `json:"source_id" gorm:"type:bigint;index:idx_source,priority:2"`
	ActionURL          string     `json:"action_url" gorm:"size:500"`
	ExtraData          string     `json:"extra_data" gorm:"type:text"`
	IncludeFutureUsers bool       `json:"include_future_users" gorm:"not null;default:false"`
	Status             int        `json:"status" gorm:"type:tinyint;not null;default:2;index:idx_audience_status_publish,priority:2;index:idx_receiver_status_publish,priority:2"`
	PublishedAt        *time.Time `json:"published_at" gorm:"index:idx_audience_status_publish,priority:3;index:idx_receiver_status_publish,priority:3"`
	ExpiresAt          *time.Time `json:"expires_at" gorm:"index:idx_expires_at"`
	CreatedAt          time.Time  `json:"created_at"`
	UpdatedAt          time.Time  `json:"updated_at"`
}

func (Notification) TableName() string { return "pm_notifications" }

type UserNotificationState struct {
	UserID         int        `json:"user_id" gorm:"primaryKey"`
	NotificationID int64      `json:"notification_id" gorm:"primaryKey"`
	IsRead         bool       `json:"is_read" gorm:"not null;default:false"`
	IsHidden       bool       `json:"is_hidden" gorm:"not null;default:false"`
	ReadAt         *time.Time `json:"read_at"`
	CreatedAt      time.Time  `json:"created_at"`
	UpdatedAt      time.Time  `json:"updated_at"`
}

func (UserNotificationState) TableName() string { return "pm_user_notification_states" }

type NotificationListItem struct {
	Notification
	IsRead bool `json:"is_read" gorm:"column:is_read"`
}

File: web/app/composables/api/notification.ts
import { useApiFetch } from '~/composables/useApiFetch'

export interface NotificationItem {
  id: number
  type: string
  audience_type: string
  receiver_id?: number | null
  sender_id?: number | null
  title: string
  content: string
  icon_name: string
  icon_color: string
  source_type?: string | null
  source_id?: number | null
  action_url?: string | null
  extra_data?: string | null
  include_future_users: boolean
  status: number
  is_read: boolean
  published_at?: string | null
  expires_at?: string | null
  created_at?: string
  updated_at?: string
}

export interface NotificationListResponse {
  list: NotificationItem[]
  total: number
  unread: number
}

export const getNotifications = (page = 1, pageSize = 20) => {
  return useApiFetch<{ code: number; msg: string; data: NotificationListResponse }>('/notification/list', {
    method: 'GET',
    query: { page, pageSize },
  })
}

export const getNotificationUnreadCount = () => {
  return useApiFetch<{ code: number; msg: string; data: { unread: number } }>('/notification/unread-count', {
    method: 'GET',
    suppressErrorMessage: true,
  })
}

export const markNotificationRead = (id: number) => {
  return useApiFetch(`/notification/${id}/read`, { method: 'PUT' })
}

export const markAllNotificationsRead = () => {
  return useApiFetch('/notification/read-all', { method: 'PUT' })
}
